src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -501,16 +501,28 @@ public function parents()
     /**
      * Returns the children nodes of the current selection.
      *
+     * @param string|null $selector An optional CSS selector to filter children
+     *
      * @return self
      *
      * @throws \InvalidArgumentException When current node is empty
+     * @throws \RuntimeException         If the CssSelector Component is not available and $selector is provided
      */
-    public function children()
+    public function children(/* string $selector = null */)
     {
+        $selector = 0 < \func_num_args() ? func_get_arg(0) : null;
+
         if (!$this->nodes) {
             throw new \InvalidArgumentException('The current node list is empty.');
         }
 
+        if (null !== $selector) {
+            $converter = $this->createCssSelectorConverter($this->isHtml);
+            $xpath = $converter->toXPath($selector, 'child::*/');
+
+            return $this->filterXPath($xpath);
```

Comment:
this could be made much more efficient by using `child::` as prefix when converting CSS and then using `filterRelativeXPath` instead of going through `relativize` to modify the XPath again